Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -1782,7 +1787,39 @@ def splunk_search_command(service):
 
     query = build_search_query(args)
     search_kwargs = build_search_kwargs(args)
-    search_job = service.jobs.create(query, **search_kwargs)  # type: ignore
+    job_sid = args.get("sid", None)
```

Comment:
The new code has made the function too long and complex in my opinion.
Let's split it wherever it makes sense, extract code to helper functions, and keep things on one level of abstraction

Reply:
done.